src/persist-types/src/arrow.rs

```diff
@@ -598,7 +598,9 @@ impl ArrayBound {
             return Some(proto);
         }
 
-        maybe_trim_proto(&mut proto, max_len);
+        let mut data_type = proto.data_type.take()?;
+        maybe_trim_proto(&mut data_type, &mut proto, max_len);
+        proto.data_type = Some(data_type);
```

Comment:
Having to remove the `data_type` feels pretty load bearing. Can you either add a `/// # Panics` section to `fn maybe_trim_proto(...)` or maybe a separate function or method on `ProtoArrayData` that pulls the `data_type` out for us?

Reply:
Sure - I'll add it to the comment!

(`maybe_trim_proto` is an implementation detail of this method, so I don't feel bad that it has this restriction in general. But since it's no longer inlined here it's best to be explicit!)